charts/tfy-configs/files/cicd-templates/gitlab-pipeline-git-source.yaml

```diff
@@ -0,0 +1,46 @@
+name: GitLab pipeline integrations to deploy new changes to TrueFoundry.
+cicd_provider_id: gitlab
+enabled: true
+description: "TrueFoundry Control Plane will git clone the source code and build the image to deploy the application."
+deployment_mode: deploy
+build_source: git
+image_builder: truefoundry-control-plane
+steps:
+  - label: Generate API Key
+    icon: null
+    usage: Generate an API Key to authenticate and deploy applications
+    type: generate-api-key
+  - label: Add API Key to Secrets
+    icon: null
+    usage: null
+    type: markdown-content
+    args:
+      content: |
+        In your GitLab Repository, navigate to **Settings > CI/CD** then expand **Variables**.
+        Add a new secret called `TFY_API_KEY` and set the generated api key as value and select the **Mask Variable** checkbox.
+  - label: Create GitLab Pipeline
+    icon: null
+    usage: |
+      Add the below workflow as `.gitlab-ci.yml` in your root directory.
+      Following Pipelines will be triggered on each push to `main` branch
+    type: markdown-content
+    args:
+      content: |
+        ```yaml
+        default:
+          image: python:3.8.19
+          
+        variables:
+          TFY_HOST: {{ TRUEFOUNDRY_TFY_HOST }}
+          TFY_APPLICATION_FQN: {{ TRUEFOUNDRY_WORKSPACE_FQN }}
+          TFY_API_KEY: $TFY_API_KEY
+          
+        deploy-job:
+          stage: deploy
+          rules:
+            - if: $CI_COMMIT_BRANCH == "main"
+          script: 
+            - pip3 install -U "truefoundry"
+            - |
+              tfy patch-application --application_fqn=tfy-ctl-euwe1-devtest:mitanshu-test-ws:emotion-class-svc --patch='{"image": {"build_source": {"ref": "'$CI_COMMIT_SHA'"}}}'
```

Comment:
This is not correct, please check the equivalent github actions file for git source
we want to use the local truefoundry.yaml we are making the user download